packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -489,13 +543,90 @@ class SliderThemeData extends Diagnosticable {
     properties.add(DiagnosticsProperty<Color>('overlayColor', overlayColor, defaultValue: defaultData.overlayColor, level: DiagnosticLevel.debug));
     properties.add(DiagnosticsProperty<Color>('valueIndicatorColor', valueIndicatorColor, defaultValue: defaultData.valueIndicatorColor));
+    properties.add(DiagnosticsProperty<SliderTrackShape>('trackShape', trackShape, defaultValue: defaultData.trackShape, level: DiagnosticLevel.debug));
+    properties.add(DiagnosticsProperty<SliderTickShape>('tickShape', tickShape, defaultValue: defaultData.tickShape, level: DiagnosticLevel.debug));
     properties.add(DiagnosticsProperty<SliderComponentShape>('thumbShape', thumbShape, defaultValue: defaultData.thumbShape, level: DiagnosticLevel.debug));
+    properties.add(DiagnosticsProperty<SliderComponentShape>('overlayShape', overlayShape, defaultValue: defaultData.overlayShape, level: DiagnosticLevel.debug));
     properties.add(DiagnosticsProperty<SliderComponentShape>('valueIndicatorShape', valueIndicatorShape, defaultValue: defaultData.valueIndicatorShape, level: DiagnosticLevel.debug));
     properties.add(EnumProperty<ShowValueIndicator>('showValueIndicator', showValueIndicator, defaultValue: defaultData.showValueIndicator));
     properties.add(DiagnosticsProperty<TextStyle>('valueIndicatorTextStyle', valueIndicatorTextStyle, defaultValue: defaultData.valueIndicatorTextStyle));
   }
 }
 
+/// Base class for slider track shapes.
+///
+/// Create a subclass of this if you would like a custom track shape.
+///
+/// See also:
+///
+///  * [RectangularSliderTrackShape] for the default track shape.
+abstract class SliderTrackShape {
+  /// Abstract const constructor. This constructor enables subclasses to provide
+  /// const constructors so that they can be used in const expressions.
+  const SliderTrackShape();
+
+  /// Returns the preferred bounds of the shape, based on the given conditions.
+  Rect getPreferredRect({
+    RenderBox parentBox,
+    Offset offset = Offset.zero,
+    SliderThemeData sliderTheme,
+    bool isEnabled,
+    bool isDiscrete,
+  });
+
+  /// Called to paint a track.
+  void paint(PaintingContext context, Offset offset, {
+    RenderBox parentBox,
+    SliderThemeData sliderTheme,
+    Animation<double> enableAnimation,
+    List<double> trackDividers,
+    List<bool> trackSegmentsActive,
+    bool isDiscrete,
+    bool isEnabled,
+  });
+}
+
+/// Base class for slider tick shapes.
+///
+/// Create a subclass of this if you would like a custom slider tick shape.
+/// This is a simplified version of [SliderComponentShape] with a
+/// [SliderThemeData] passed when getting the preferred size.
+///
+/// See also:
+///
+///  * [RoundSliderTickShape] for a simple example of a tick shape.
+abstract class SliderTickShape {
+  /// Abstract const constructor. This constructor enables subclasses to provide
```

Comment:
Nit: Same as above (first sentence is own paragraph).

Reply:
See related comment on  #25672

 Since this was copied from the current pattern, I am going to leave it as is, then update the docs in a follow up PR to make them consistent